install-monorepo.sh

```diff
@@ -0,0 +1,117 @@
+OS="$(uname)"
+if [[ "$OS" == "Linux" ]]; then
+	ON_LINUX=1
+	sudo apt install build-essential
+elif [[ "$OS" != "Darwin" ]]; then
+	abort "Installer script is only supported on macOS and Linux."
+fi
+
+# Check of Homebrew and nvm are installed
+echo "Checking if Homebrew is installed..."
+if [[ -z "$(command -v brew)" ]]; then
+    echo "Installing Homebrew"
+    /bin/bash -c "$(curl -fsSL https://raw.githubusercontent.com/Homebrew/install/HEAD/install.sh)"
+	if [[ ON_LINUX ]]; then # Add homebrew to PATH
+		echo 'eval "$(/home/ubuntu/.linuxbrew/bin/brew shellenv)"' >> /home/ubuntu/.profile
+		eval "$(/home/linuxbrew/.linuxbrew/bin/brew shellenv)"
+		PATH=/home/linuxbrew/.linuxbrew/bin:$PATH
+		hash -r 
+	fi
+else
+	echo "Updating brew"
+    brew update
+fi
+
+echo "Checking if NVM is installed..."
+if [[ -z "$(command -v nvm)" ]]; then
+    echo "Installing nvm"
+    curl -o- https://raw.githubusercontent.com/nvm-sh/nvm/v0.39.0/install.sh | bash && export NVM_DIR=$HOME/.nvm && source $NVM_DIR/nvm.sh
+else
+	echo "Updating nvm"
+    nvm update
+fi
+
+# Install and use the correct version of Node.js
+echo "Installing Node.js"
+nvm install && nvm use
+
+# Install our requirements
+echo "Installing Bash"
+brew install bash
+
+echo "Checking if jq is installed..."
+if [[ -z "$(command -v jq)" ]]; then
+	echo "Installing jq"
+	brew install jq
+fi
+
+echo "Checking if pnmp is installed..."
+if [[ -z "$(command -v pnpm)" ]]; then
+	echo "Installing pnpm"
+	curl -f https://get.pnpm.io/v6.16.js | node - add --global pnpm
+fi 
+
+echo "Installing and linking PHP 8.0"
+source .github/versions.sh && brew install php@$PHP_VERSION
+brew link php@8.0
+
+echo "Checking composer..."
+if [[ -z "$(command -v composer)" ]]; then
+echo "Installing Composer"
+	EXPECTED_CHECKSUM="$(php -r 'copy("https://composer.github.io/installer.sig", "php://stdout");')"
+	php -r "copy('https://getcomposer.org/installer', 'composer-setup.php');"
+	ACTUAL_CHECKSUM="$(php -r "echo hash_file('sha384', 'composer-setup.php');")"
+
+	if [ "$EXPECTED_CHECKSUM" != "$ACTUAL_CHECKSUM" ]
+	then
+		>&2 echo 'ERROR: Invalid installer checksum'
+		rm composer-setup.php
+		exit 1
+	fi
+
+	php composer-setup.php --version=2.1.8 --quiet
+	RESULT=$?
+	rm composer-setup.php
+	echo "$RESULT"
+	sudo mv composer.phar /usr/local/bin/composer
+fi 
+
+# Setup the Jetpack CLI
+echo "Setting up the Jetpack CLI"
+pnpm install && pnpm cli-setup
+pnpm jetpack cli link # I don't know why we have to do this twice, but it works.
+jetpack install --root
+
+# Reset the terminal so it picks up the changes.
+if [[ "$SHELL" == "/bin/zsh" ]]; then
+	echo "Refreshing terminal"
+	exec zsh
+elif [[ "$SHELL" == "/bin/bash" ]]; then
+	echo "Refreshing terminal"
+	exec bash
+else 
+	echo "Note: You may have to restart your terminal for monorepo tools to work properly."
+fi
+
+echo "Installation complete. You may run tools/check-development-environment.sh to make sure everything installed correctly"
```

Comment:
You'll never get here if lines 109 or 112 are reached.

Also, there should be a newline at EOF.

Reply:
Yeah, this is what I struggled with the most. Do you have a recommendation for how to exit the script without having to tell the user to restart their terminal or run `source ./bashrc` or some such? I could get the installer to work, but without `exec bash` etc, things wouldn't work right away without further input.

Reply:
There's no way to do it, as far as I know. Telling them to close and reopen the terminal is probably the most straightforward, as even if you do something here they might have other windows open that would also need restarting if they want to make use of stuff.

Reply:
What do you think about leaving the `exec bash` etc code, but then adding the `Installation complete. You may need to restart your terminal for changes to take effect, then run tools/check-development-environment.sh to make sure everything is installed correctly.`  ? Or would you remove it altogether?

Reply:
I suppose the question then is whether we want the current terminal to work, but then if they `exit` or Ctrl+D it doesn't seem to exit (but returns them to a potentially not-working shell), versus basically requiring they close and reopen the terminal. I don't really have an opinion on that.

Reply:
I just moved it for now and we can adjust it later if it's causing problems.